backend/tasks/user.py

```diff
@@ -0,0 +1,54 @@
+from flask_restful import Resource, reqparse
+from flask import request
+from psycopg2.extras import RealDictCursor
+from pymysql.cursors import DictCursor
+import os
+from flask import make_response
+import json
+import logging
+from flask_login import login_required
+logger = logging.getLogger(__name__)
+
+
+class UserTask(Resource):
+
+    @login_required
+    def get(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument('search', type=str, location='args')
+        args = parser.parse_args()
+        search_param = args.get('search')
+        if search_param is None or len(search_param) == 0:
+            return make_response(json.dumps({'message': 'Parameter \'search\' is required'}), 400)
+
+        try:
+            with request.db.cursor(cursor_factory=RealDictCursor) if os.environ.get('IS_LOCAL') else request.db.cursor(DictCursor) as cur:
+                cur.execute(f"SELECT \"user\".username, \"user\".enabled FROM \"user\" WHERE \"user\".username LIKE '%{search_param}%'")
+                result = cur.fetchall()
+                request.db.commit()
+                return make_response(json.dumps(result), 200)
+        except Exception as e:
+            logger.exception("Unable to fetch users: " + str(e))
+            return make_response(json.dumps({'message': 'Unable to fetch users'}), 500)
+
+    # Used for enabling a user, under the general put function
+    # Trying to keep this as RESTful as possible, but may want to separate out into it's own 'enable' endpoint
+    @login_required
+    def put(self):
```

Comment:
Ideally when disabling a user, you would also cleanup the resources related to his repos:
- builds
- triggers
- mirrors
- queue items

And all this should be wrapped in a single transaction.
See `util/disableabuser.py`

Reply:
Okay let me add this in.